lib/install/ova/configure.go

```diff
@@ -0,0 +1,117 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ova
+
+import (
+	"context"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+	virtualmachine "github.com/vmware/vic/pkg/vsphere/vm"
+)
+
+const (
+	// ManagedByKey defines the extension key to use in the ManagedByInfo of the OVA
+	ManagedByKey = "com.vmware.vic"
+	// ManagedByType defines the type to use in the ManagedByInfo of the OVA
+	ManagedByType = "VicApplianceVM"
+)
+
+func configureManagedByInfo(ctx context.Context, sess *session.Session, v *object.VirtualMachine) error {
+	vm := virtualmachine.NewVirtualMachineFromVM(ctx, sess, v)
+	spec := types.VirtualMachineConfigSpec{
+		ManagedBy: &types.ManagedByInfo{
+			ExtensionKey: ManagedByKey,
+			Type:         ManagedByType,
+		},
+	}
+
+	info, err := vm.WaitForResult(ctx, func(ctx context.Context) (tasks.Task, error) {
+		return vm.Reconfigure(ctx, spec)
+	})
+
+	if err != nil {
+		log.Errorf("Error while setting ManagedByInfo: %s", err)
+		return err
+	}
+
+	if info.State != types.TaskInfoStateSuccess {
+		log.Errorf("Setting ManagedByInfo reported: %s", info.Error.LocalizedMessage)
+		return err
+	}
+
+	return nil
+}
+
+// ConfigureManagedByInfo takes sets the ManagedBy field for the VM specified by ovaURL
+func ConfigureManagedByInfo(ctx context.Context, sess *session.Session, ovaURL string) error {
+	vm, err := getVM(ctx, sess.Vim25(), ovaURL)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Attempting to configure ManagedByInfo")
+	err = configureManagedByInfo(ctx, sess, vm)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Successfully configured ManagedByInfo")
+	return nil
+}
+
+func getVM(ctx context.Context, client *vim25.Client, u string) (*object.VirtualMachine, error) {
+	ovaURL, err := url.Parse(u)
+	if err != nil {
+		return nil, err
+	}
+
+	ip := ovaURL.Hostname()
+
+	finder := find.NewFinder(client, true)
+	dc, err := finder.Datacenter(ctx, "*")
+	if err != nil {
+		return nil, err
+	}
+
+	searchIndex := object.NewSearchIndex(client)
+	ref, err := searchIndex.FindByIp(ctx, dc, ip, true)
+	if err != nil {
+		return nil, errors.Errorf("failed to find vm by IP: %s", err)
+	}
+
+	if ref == nil {
+		return nil, errors.Errorf("failed to find vm with IP %s", ip)
+	}
+
+	ref, err = finder.ObjectReference(ctx, ref.Reference())
```

Comment:
I suggest checking that the VM has the guestinfo keys we expect - this would protect against accidentally identifying the wrong VM (for example if multiple VMs have the same IP which is perfectly acceptable and perhaps even expected with NSX/VXLAN configurations).
Although it looks like `FindByIp` only returns a single match...

Reply:
Fixed, let me know if you have feedback on my approach!